python/tests/operator/test_evolve_dynamics.py

```diff
@@ -0,0 +1,660 @@
+                              rho0,
+                              observables=model.observables(),
+                              collapse_operators=model.collapse_ops(),
+                              store_intermediate_results=True,
+                              integrator=integrator())
+    # Check expectation values
+    for i, exp_vals in enumerate(evolution_result.expectation_values()):
+        np.testing.assert_allclose([res.expectation() for res in exp_vals],
+                                   model.expect(model.tlist[i]),
+                                   atol=1e-3)
+    # Check intermediate states
+    for i, state in enumerate(evolution_result.intermediate_states()):
+        np.testing.assert_allclose(state,
+                                   model.expected_state(model.tlist[i]),
+                                   atol=1e-3)
+
+@pytest.mark.parametrize('integrator', all_integrator_classes)
+def test_floquet_steady_state(integrator):
+    # two-level system coupled with the external heat bath (fixed temperature)
+    # (time-dependent Hamiltonian and multiple collapse operators)
+    delta = (2 * np.pi) * 0.3
+    eps_0 = (2 * np.pi) * 1.0
+    A = (2 * np.pi) * 0.05
+    w = (2 * np.pi) * 1.0
+    kappa_1 = 0.15
+    kappa_2 = 0.05
+    sx = spin.x(0)
+    sz = spin.z(0)
+    sm = operators.annihilate(0)
+    sm_dag = operators.create(0)
+    dimensions = {0: 2}
+
+    # Hamiltonian
+    hamiltonian = -delta / 2.0 * sx - eps_0 / 2.0 * sz + A / 2.0 * ScalarOperator(
+        lambda t: np.sin(w * t)) * sz
+
+    psi0 = cudaq.State.from_data(cp.array([1.0, 0.0], dtype=cp.complex128))
+
+    # Thermal population
+    n_th = 0.5
+
+    # Collapse operators
+    c_op_list = [
+        np.sqrt(kappa_1 * (1 + n_th)) * sm,
+        np.sqrt(kappa_1 * n_th) * sm_dag,
+        np.sqrt(kappa_2) * sz
+    ]
+
+    # Schedule of time steps.
+    steps = np.linspace(0, 50, 500)
+    schedule = Schedule(steps, ["t"])
+
+    # Run the simulation.
+    # First, we run the simulation without any collapse operators (no decoherence).
+    evolution_result = evolve(hamiltonian,
+                              dimensions,
+                              schedule,
+                              psi0,
+                              observables=[operators.number(0)],
+                              collapse_operators=c_op_list,
+                              store_intermediate_results=True,
+                              integrator=integrator())
+
+    expected_steady_state = 0.27
+    n = [
+        exp_vals[0].expectation()
+        for exp_vals in evolution_result.expectation_values()
+    ]
+
+    # should converge to the steady-state after half of the duration
+    for i in range(len(n) // 2, len(n)):
+        # Note: the final population is still oscillating.
+        np.testing.assert_allclose(n[i], expected_steady_state, atol=0.02)
+
+
+@pytest.mark.parametrize('integrator', all_integrator_classes)
+def test_landau_zener(integrator):
```

Comment:
This is an example taken from https://github.com/qutip/qutip-notebooks/blob/master/examples/landau-zener.ipynb

Reply:
I disagreed with the statement that this is a taken example. The above reference notebook was written as a tutorial about the basic Landau-Zener physics in QuTiP. The same physics can also be expressed in CUDA-Q dynamics as shown in this example.

Reply:
All the parameters and parameter names are identical and even the ranges and number of steps for the time evolution match exactly.